### DISCUSSION TITLE
New beta version 1.0.8-beta4

### DISCUSSION THREAD

**Discussion by pierreguillot** [pierreguillot/Camomile#232] [Announcement]
Hi,

I published a new temporary version [1.0.8-beta4](https://github.com/pierreguillot/Camomile/releases/tag/v1.0.8-beta4). This beta version should be stable (at least not less stable than the release 1.0.7) but I didn't manage to do all the features and fixes that I wanted to do for the version [1.0.8](https://github.com/pierreguillot/Camomile/projects/12) yet. You can find the list of the fixes and improvements that are already done on the release section.

Your feedback would be much appreciated

**Comment by pierreguillot** [comment]
And here is a little demo of live patching in Reaper with Camomile and the object pd~:

https://user-images.githubusercontent.com/1409918/104299893-0c352780-54c6-11eb-924b-4d5bbc903b81.mp4

**Reply by stefanzintel** [reply]
WOW, looking forward to use it in Logic Pro X.
I tried to download it. unfortunately only have the 1.07. is there also a download link for the 1.08 beta version? OSX / AU? Thanks

**Reply by pierreguillot** [reply]
You can use the first link of the [first message of this discussion](https://github.com/pierreguillot/Camomile/discussions/232#discussion-1720885).

**Reply by stefanzintel** [reply]
I downloaded exactly this file ... unfortunately it shows up as 1.07 after the installation.

**Reply by pierreguillot** [reply]
I didn't update the version number yet.

**Reply by micheldejoode** [reply]
Hi, this live patching demo looks great! Can you explain how to do this? Do you have an example PD file?

**Reply by pierreguillot** [reply]
Yes, as I wrote [here](https://github.com/pierreguillot/Camomile/discussions/233#discussioncomment-309313), I'll try to create a new section on the wiki with some examples soon. But I can also start to share it here, this way I could get your feedback. I'm pretty busy but I'll do it as soon as possible.

**Reply by solipd** [reply]
pls share if you have the time. I couldn't get any sound from the pd-subproces in reaper on linux

**Reply by micheldejoode** [reply]
Hi Pierre, I would by happy to test some examples! But I also understand that you're busy, so I will be patient :-)

**Reply by solipd** [reply]
> I couldn't get any sound from the pd-subproces in reaper on linux

my mistake, it works! great feature, thanks for all the work

**Reply by micheldejoode** [reply]
Hi solipd, i'm rather new to camomile, but found out how the pd~ object works and made an example of it. But when I make a VST from it using camomile and load it in my daw (Reaper), in the camomile console it says: can't find pd.exe. Any idea what i'm doing wrong here?

**Reply by alfonso73** [reply]
@micheldejoode Hi, can you share your pd patch and the camomile .txt so i can test and see if i can help?

**Comment by micheldejoode** [comment]
[PdSub.zip](https://github.com/pierreguillot/Camomile/files/5898635/PdSub.zip)
Hi alfonso73, here is my patch. It's mostly the help-patch for the pd~ object, I just want to open the subprocess, so don't look at params etc. Thanks!

**Reply by alfonso73** [reply]
thanx @micheldejoode 
basically you have to add the -pddir and -scheddir flags to [pd~ ].
I did [ pd~ -ninsig 2 -noutsig 2 -fifo 20 -pddir C:/Program\ Files/Pd/bin -scheddir C:/Program\ Files/Pd/extra/pd~ ] and so no errors in Camomile console.
I wonder how the main patch trigger would behave with an [loadbang] triggering it. Because i imagine to have i.e. 10 or 20 VST3 Camomile and have to open all of them manually :-/...and then close all the pd~ windows....kinda messy ;-)

**Reply by micheldejoode** [reply]
Hi Alfonso73, I've got it working now with your help, thanks! Still have to look into it, I have no sound in the subpatch, but I will find out. The trigger works also with a loadbang, tested it.

**Comment by alfonso73** [comment]
Yeah. Great. You should add some audio in/out in the sub patch. Yeah
loadbang does work. The only "heavy" drawback is that with 20 Camomile VST3
using [pd~] a 20 pd instances with a window each would be a serious screen
cluttering business. I'll do some more test and report my opinion on the
[pd~] approach.

**Reply by micheldejoode** [reply]
Hi alfonso, I've got audio working now. And yes, what to do when you have multiple pd instances... It would be nice if all instances were in tabs like in a browser or something. I also thought about a switch with the -nogui option, but haven't got that working. And the 'stop' message is important to kill the instance, but that's not a solution if you want to use the instance on another moment. Anyway, some thinking and testing to do...

**Comment by pierreguillot** [comment]
Here is an [example](https://github.com/pierreguillot/Camomile/discussions/233#discussioncomment-344215) with some explanation. Your feedback is welcome!
